src/utils/colonyCoinMachine.tsx

```diff
@@ -0,0 +1,56 @@
+import { FormattedMessage, defineMessages } from 'react-intl';
```

Comment:
No, this is not ok. Util files should be `.ts` not `.tsx`.

Only pure components should be a `.tsx` file, otherwise they are not utils.

I saw you wanted to return a message descriptor, but you can also do so programatically, without involving a React component. You can import `formatMessage` from `react-intl` and use that to parse your values.

Please refactor this.